src/semver/_types.py

```diff
@@ -8,3 +9,4 @@
 VersionIterator = Iterable[VersionPart]
 String = Union[str, bytes]
 F = TypeVar("F", bound=Callable)
+DecoratorType = Union[Callable[..., F], partial]
```

Comment:
@tomschr Good idea to add this alias, we just don't need the `Type` suffix since it is clear it is a type from it's import path.
```suggestion
Decorator = Union[Callable[..., F], partial]
```